### PR TITLE
[serve] Allow route prefix override to /

### LINKED ISSUE #33918: [serve] `serve.run` ignores setting `route_prefix="/"`
### What happened + What you expected to happen

If user runs an application with `route_prefix` set to `"/"` in `serve.run()`, that route prefix will be ignored and overridden by the ingress deployment's route prefix.

The route prefix passed directly to `serve.run()` should always take priority.

### Versions / Dependencies

master

### Reproduction script

Run this:
```
@serve.deployment
def f():
   return "hello"

ingress_node = f.options(route_prefix="/hello").bind()
serve.run(ingress_node, route_prefix="/")
```

The application will be unreachable at `http://localhost:8000/` but reachable at `http://localhost:8000/hello`

### Issue Severity

None

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

`serve.run(route_prefix=...)` should override any route prefix set in code for the ingress deployment. The bug is that if you set `serve.run(route_prefix="/")`, "/" is ignored and instead the route prefix of the ingress deployment is used. This is inconsistent.

This happens because we use the default "/" in the parameters of functions, so we can't differentiate between when users explicitly set it or if it was unset and loaded by default. Setting the default parameter to DEFAULT.VALUE fixes this.

## Related issue number

Closes #33918 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/tests/test_api.py [hunk 1]
```diff
@@ -646,6 +646,31 @@ def __call__(self):
     assert requests.get("http://127.0.0.1:8000/").text == "got model"
 
 
+@pytest.mark.parametrize(
+    "ingress_route,app_route",
+    [("/hello", "/"), ("/hello", "/override"), ("/", "/override")],
```
Comment (shrekris-anyscale) [inline python/ray/serve/tests/test_api.py:651]: Could you also add tests for (`None`, `None`), (non-`None`,`None`) and (`None`, non-`None`)?

**Review by shrekris-anyscale — COMMENTED** [review]
Nice work so far! Left a couple questions.

File: python/ray/serve/api.py [hunk 2]
```diff
@@ -539,7 +539,7 @@ def run(
 
     for deployment in deployments:
         # Overwrite route prefix
-        if route_prefix != "/" and deployment._route_prefix:
+        if route_prefix is not DEFAULT.VALUE and deployment._route_prefix is not None:
```
[the remark this answers is not in the document]
Comment (zcin) [reply python/ray/serve/api.py:542]: That checking is done in `get_and_validate_ingress_deployment`.

File: python/ray/serve/tests/test_api.py (hunk 1 above)
Comment (zcin) [reply python/ray/serve/tests/test_api.py:651]: Good point, will do!

File: python/ray/serve/api.py (hunk 2 above)
Comment (sihanwang41) [inline python/ray/serve/api.py:542]: why do we check `deployment._route_prefix is not None`? if route_prefix is not default value, should always do `deployment._route_prefix = route_prefix`?

Comment (zcin) [reply python/ray/serve/api.py:542]: `deployment._route_prefix is not None` only for the ingress deployment.

File: python/ray/serve/tests/test_api.py (hunk 1 above)
Comment (zcin) [reply python/ray/serve/tests/test_api.py]: Added!

**Review by shrekris-anyscale — APPROVED** [review]
Nice work!

**Review by sihanwang41 — APPROVED** [review]

**Comment by zcin** [comment]
@edoakes ready for merge!
